File: BOJ/ksh/1149.py
n = int(input())
arr = []

for i in range(n):
    arr.append(list(map(int,input().split())))

'''
arr[i][0] = R
arr[i][1] = G
arr[i][2] = B
'''
for i in range(1, n):
    arr[i][0] = min(arr[i-1][1], arr[i-1][2]) + arr[i][0]
    arr[i][1] = min(arr[i-1][0], arr[i-1][2]) + arr[i][1]
    arr[i][2] = min(arr[i-1][0], arr[i-1][1]) + arr[i][2]

print(str(min(arr[n-1][0], arr[n-1][1], arr[n-1][2])))
print(arr[n-1][0], arr[n-1][1], arr[n-1][2])

File: BOJ/lgd/1149.cpp
#include <bits/stdc++.h>

using namespace std;

/*
*/

int main() {
	cin.tie(nullptr);
	cout.tie(nullptr);
	ios_base::sync_with_stdio(false);
		
	int distance[1000][3] = { -1 };
	int input[1000][3];
	int N;
	int result = 0;
	cin >> N;

	for (int i = 0; i < N; i++) {
		for (int j = 0; j < 3; j++) {
			cin >> input[i][j];
		}
	}
	for (int i = 0; i < 3; i++) {
		distance[0][i] = input[0][i];
	}

	for (int i = 1; i < N; i++) {
		for (int j = 0; j < 3; j++) {
			if (j == 0) {
				distance[i][j] =input[i][j] + min(distance[i-1][1], distance[i-1][2]);
			}
			else if (j == 1) {
				distance[i][j] = input[i][j] + min(distance[i - 1][0], distance[i - 1][2]);
			}
			else {
				distance[i][j] = input[i][j] + min(distance[i - 1][0], distance[i - 1][1]);
			}
		}
	}

	result = distance[N - 1][0];

	for (int i = 1; i < 3; i++) {
		if (result > distance[N - 1][i]) {
			result = distance[N - 1][i];
		}
	}

	cout << result;

	return 0;
}